pkg/controller/machine/machine_controller.go

```diff
@@ -18,13 +18,19 @@ package machine
 
 import (
 	"context"
+	"fmt"
 	"os"
+	"time"
 
+	"github.com/go-log/log/info"
+	kubedrain "github.com/openshift/kubernetes-drain"
```

Comment:
It's not likely to go stale or unmaintained because we ship it in OpenShift.  In any case, this is no different than vendoring in any number of dependencies that you see in the wild.  Here's an example of said 'non standard' type library: https://github.com/petar/GoLLRB

It's actually not pulling in an additional logging library, that library is actually just a simple interface (discussed previously).

I've looked into adding support from the kubectl bits as well, we decided against that already if you check out the resolved conversations.

If there's another library you can suggest, I'm happy to use that instead.  Or we could vendor the code here so it doesn't go stale.  Or we can be good consumers and consume from the community and contribute fixes back.

Reply:
How hard would it be to vendor both the cmd and library portions from k/k?

Reply:
@ncdc I looked into that as well.  The cmd portion is the tricky part, it imports a lot of other things from k/k.  So, it would need a total refactor.  I started down this road, but it quickly became a mess.

I think we should either use it as-is, or we should copy the code into cluster-api.  I'm in favor of the former, and then when the kubectl stuff gets pulled into it's own project, we can consume that instead.

Reply:
@ncdc I confirm this is a bug.  We're not actually waiting for the pod to be deleted it seems, and this is not what we want.